src/dict.c

```diff
@@ -703,9 +727,17 @@ dictEntry *dictUnlink(dict *d, const void *key) {
  * to dictUnlink(). It's safe to call this function with 'he' = NULL. */
 void dictFreeUnlinkedEntry(dict *d, dictEntry *he) {
     if (he == NULL) return;
+    _dictFreeUnlinkedEntry(d, he);
+}
+
+static void _dictFreeUnlinkedEntry(dict *d, dictEntry *he) {
```

Comment:
Why extract it? feels like using dictFreeUnlinkedEntry directly is the same.

Reply:
Undone in the latest version.